cspell.json

```diff
@@ -0,0 +1,81 @@
+{
+  "language": "en",
+  "words": [
+    "Blurhash",
+    "bryc",
+    "bufferutil",
+    "Cathelineau",
+    "clsx",
+    "Codespaces",
+    "corejs",
+    "cyrb",
+    "darkred",
+    "debouncee",
+    "debouncer",
+    "debouncers",
+    "deduplication",
+    "Delapouite",
+    "deps",
+    "Dyno",
+    "Faithtoken",
+    "favicons",
+    "ffprobe",
+    "fontawesome",
+    "fortawesome",
+    "frameloop",
+    "freesound",
+    "GLTF",
+    "Immer",
+    "inotify",
+    "ispending",
+    "jcalz",
+    "Lerped",
+    "lerping",
+    "linkify",
+    "lorc",
+    "namegen",
+    "noreferrer",
+    "Nygård",
+    "pathinfo",
+    "pmmmwh",
+    "polymorphed",
+    "Preet",
+    "Prefetch",
+    "Promisable",
+    "reduxjs",
+    "roughjs",
+    "RRID",
+    "sbed",
+    "seznam",
+    "Shihn",
+    "sjson",
+    "Skoll",
+    "slerp",
+    "stndrdth",
+    "streetsidesoftware",
+    "Svish",
+    "tada",
+    "tailwindcss",
+    "tinycolor",
+    "tippyjs",
+    "typanion",
+    "unfav",
+    "unnotified",
+    "unstyled",
+    "vcxsrv",
+    "xvfb"
+  ],
+  "ignorePaths": [
+    "package-lock.json",
+    "node_modules",
+    "vscode-extension",
+    ".git/objects",
+    ".vscode",
+    ".vscode-insiders",
+    "package.json",
+    "yarn.lock",
+    "main.yml",
+    ".gitignore",
+    ".eslintrc.json"
+  ]
```

Comment:
While playing around with this locally, I noticed that we are enforcing spelling in way to many files :D 
- `lint-staged` is configured to run on every changed file, which means that it would eventually call `cspell` for every single file that is part of the GitHub repo if you were to touch all of the files in one commit. This includes binary files (.svg, .mp3, ...) and hidden files (files and directories starting with `.`).
- The GitHub action also [only checks changed files](https://github.com/streetsidesoftware/cspell-action#usage), but would run into the same problem when we ever touch or add new binary files
- The VSCode extension probably also happily checks binary files for spelling errors at the moment

(In all cases, CSpell will ignore files listed in `ignorePaths` (expected), as well as all hidden files (perhaps unexpected?))

I think we should make sure that we correctly allowlist or denylist the set of files we actually want to check spelling for. I feel like an allowlist approach might be easiest, least frustrating, and most viable option for us. The allowlist should probably be part of `cspell.json` by configuring the `files` option (the documentation for that option doesn't really exist, but it appears to work). We could, for instance, use something like this, but that's just an idea:

```jsonc
"files": [
  "**/*.{js,jsx,ts,tsx,html,md}" // add css,scss? peggy? json? yml?
],
"ignorePaths": [
  "third-party" // not sure?
]
```

To confirm that it works, run `yarn cspell lint` and check that there are no errors.
To also scan hidden files (could be a good idea, I think), we can enable the [`enableGlobDot` option](https://github.com/streetsidesoftware/cspell/pull/1985).